src/model/test/HeatPumpPlantLoopEIRCooling_GTest.cpp

```diff
@@ -0,0 +1,157 @@
+    hp.electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve();
+  EXPECT_TRUE(electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve);
+}
+
+TEST_F(ModelFixture, HeatPumpPlantLoopEIRCooling_GettersSetters) {
+  Model m;
+  HeatPumpPlantLoopEIRCooling hp(m);
+
+  EXPECT_TRUE(hp.setCondenserType("AirSource"));
+  HeatPumpPlantLoopEIRHeating companionHP(m);
+  EXPECT_TRUE(hp.setCompanionHeatingHeatPump(companionHP));
+  EXPECT_TRUE(hp.setReferenceLoadSideFlowRate(1.0));
+  EXPECT_TRUE(hp.setReferenceSourceSideFlowRate(2.0));
+  EXPECT_TRUE(hp.setReferenceCapacity(3.0));
+  EXPECT_TRUE(hp.setReferenceCoefficientofPerformance(4.0));
+  EXPECT_TRUE(hp.setSizingFactor(5.0));
+  CurveBiquadratic curve1(m);
+  EXPECT_TRUE(hp.setCapacityModifierFunctionofTemperatureCurve(curve1));
+  CurveBiquadratic curve2(m);
+  EXPECT_TRUE(hp.setElectricInputtoOutputRatioModifierFunctionofTemperatureCurve(curve2));
+  CurveQuadratic curve3(m);
+  EXPECT_TRUE(hp.setElectricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve(curve3));
+
+  EXPECT_EQ("AirSource", hp.condenserType());
+  ASSERT_TRUE(hp.companionHeatingHeatPump());
+  EXPECT_EQ(companionHP.handle(), hp.companionHeatingHeatPump().get().handle());
+  ASSERT_TRUE(hp.referenceLoadSideFlowRate());
+  EXPECT_EQ(1.0, hp.referenceLoadSideFlowRate().get());
+  EXPECT_FALSE(hp.isReferenceLoadSideFlowRateAutosized());
+  ASSERT_TRUE(hp.referenceSourceSideFlowRate());
+  EXPECT_EQ(2.0, hp.referenceSourceSideFlowRate());
+  EXPECT_FALSE(hp.isReferenceSourceSideFlowRateAutosized());
+  ASSERT_TRUE(hp.referenceCapacity());
+  EXPECT_EQ(3.0, hp.referenceCapacity().get());
+  EXPECT_FALSE(hp.isReferenceCapacityAutosized());
+  EXPECT_EQ(4.0, hp.referenceCoefficientofPerformance());
+  EXPECT_EQ(5.0, hp.sizingFactor());
+  EXPECT_EQ(curve1.handle(), hp.capacityModifierFunctionofTemperatureCurve().handle());
+  EXPECT_EQ(curve2.handle(), hp.electricInputtoOutputRatioModifierFunctionofTemperatureCurve().handle());
+  EXPECT_EQ(curve3.handle(), hp.electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve().handle());
+
+  hp.autosizeReferenceLoadSideFlowRate();
+  hp.autosizeReferenceSourceSideFlowRate();
+  hp.autosizeReferenceCapacity();
+
+  EXPECT_TRUE(hp.isReferenceLoadSideFlowRateAutosized());
+  EXPECT_TRUE(hp.isReferenceSourceSideFlowRateAutosized());
+  EXPECT_TRUE(hp.isReferenceCapacityAutosized());
+}
+
+TEST_F(ModelFixture, HeatPumpPlantLoopEIRCooling_remove) {
+  Model m;
+  HeatPumpPlantLoopEIRCooling hp(m);
+  auto size = m.modelObjects().size();
+  EXPECT_FALSE(hp.remove().empty());
+  EXPECT_EQ(size - 1, m.modelObjects().size());
+  EXPECT_EQ(0u, m.getConcreteModelObjects<HeatPumpPlantLoopEIRCooling>().size());
+}
+
+TEST_F(ModelFixture, HeatPumpPlantLoopEIRCooling_clone) {
+  Model m;
+  CurveBiquadratic curve1(m);
+  CurveBiquadratic curve2(m);
+  CurveQuadratic curve3(m);
+  HeatPumpPlantLoopEIRCooling hp(m, curve1, curve2, curve3);
```

Comment:
```suggestion
  HeatPumpPlantLoopEIRCooling hp(m, curve1, curve2, curve3);
  EXPECT_EQ(3u, m.getConcreteModelObjects<Curve>().size());
```

Reply:
@jmarrec Turns out `Curve` isn't a concrete model object, so I don't thin this will work. Instead I replaced with:
```
EXPECT_EQ(2u, m.getConcreteModelObjects<CurveBiquadratic>().size());
EXPECT_EQ(1u, m.getConcreteModelObjects<CurveQuadratic>().size());
```

Reply:
Sure that works. m.getModelObjects<Curve>() would have worked too.

getConcreteModelObjects is preferred for **Concrete** ones because it's faster.

https://github.com/NREL/OpenStudio/blob/dd40326c4deeb34e3af2641535ffa4f65149db2f/src/model/Model.hpp#L288-L323